### DISCUSSION TITLE
anyone knows is there a website that explains the syntax used to comment apis/functions in a lot of js docs?

### DISCUSSION THREAD

**Discussion by devhandler** [supabase/supabase#16802] [Questions] [answered]
i searched a lot. for example, the below uses <T>, which seems come from Java's generic type. but this is js, why using java syntax of comment? and all these special notation like ?, |, [], are hard to understand what they mean exactly. does anyone know that if there is such a website explaining those comment syntax?

```
function writable<T>(
  value?: T | undefined,
  start?: [StartStopNotifier](https://svelte.dev/docs/svelte-store#types-startstopnotifier)<T> | undefined
): [Writable](https://svelte.dev/docs/svelte-store#types-writable)<T>;
```

**Comment by GaryAustin1** [comment, marked as answer]
It is Typescript.
https://www.typescriptlang.org/docs/handbook/2/types-from-types.html
